openedx/features/enterprise_support/utils.py

```diff
@@ -401,7 +402,15 @@ def is_enterprise_learner(user):
     Returns:
         (bool): True if given user is an enterprise learner.
     """
-    return EnterpriseCustomerUser.objects.filter(user_id=user.id).exists()
+    cached_is_enterprise_key = '{}:user_is_enterprise'.format(user.id)
+    if cache.get(cached_is_enterprise_key):
+        return True
+
+    if EnterpriseCustomerUser.objects.filter(user_id=user.id).exists():
+        # Cache the enterprise user for one hour.
+        cache.set(cached_is_enterprise_key, True, expiry=3600)
+        return True
+    return False
```

Comment:
....and update the test for `is_enterprise_learner()` to cover this new caching logic.